test/extensions/common/aws/mocks.h

```diff
@@ -12,19 +12,14 @@ namespace Aws {
 
 class MockCredentialsProvider : public CredentialsProvider {
 public:
-  MockCredentialsProvider();
-  ~MockCredentialsProvider() override;
```

Comment:
Please revert this change. This is a perf optimization recommended by gmock.

Reply:
I'll revert it. But I couldn't find anything in the gtest/gmock documentation that speak to that guideline. 
Please point me to it if you can. I'm curious.

Reply:
https://github.com/google/googletest/blob/master/googlemock/docs/cook_book.md#making-the-compilation-faster